### PR TITLE
Resolve JBUGs in err.bal

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Related to #550

### CODE REVIEW

File: compiler/modules/err/err.bal [hunk 1]
```diff
@@ -37,11 +37,9 @@ public type Detail record {
 
 public type Any Syntax|Semantic|Unimplemented;
 
-// Grammatical inconsistency here is reluctantly intentional
```
Comment (jclark) [inline compiler/modules/err/err.bal:40]: Don't remove this line

File: compiler/modules/err/err.bal [hunk 2]
```diff
@@ -37,11 +37,9 @@ public type Detail record {
...
-// JBUG defining these as `error<Detail> & distinct error` results in these errors not being distinct
-public type Syntax distinct error<Detail>;
```
Comment (jclark) [inline compiler/modules/err/err.bal:42]: No need to change these lines. Just remove the JBUG line

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/err/err.bal (hunk 1 above)
Comment (KavinduZoysa) [reply compiler/modules/err/err.bal:40]: Added.

File: compiler/modules/err/err.bal (hunk 2 above)
Comment (KavinduZoysa) [reply compiler/modules/err/err.bal:42]: Changed

**Pull request merged**